src/lib/Makefile.am

```diff
@@ -9,7 +9,7 @@ pkgconfig_DATA = ddccontrol.pc
 
 $(pkgconfig_DATA): $(srcdir)/ddccontrol.pc.in $(top_builddir)/config.status
 
-modulesdir = /etc/modules-load.d/
+modulesdir = $(libdir)/modules-load.d/
```

Comment:
Note the distinction between `$(libdir)` and `$(prefix)/lib`. Systemd expects the latter https://github.com/systemd/systemd/blob/6b87254ad1ecb17e4ef88090c356705e55088c8a/src/core/systemd.pc.in#L77

Reply:
@jtojnar Hmm, I see your point. From the man page the OP links to, the locations systemd will expect to find the kernel module configs from are:

```
 /etc/modules-load.d/*.conf
/run/modules-load.d/*.conf
/usr/lib/modules-load.d/*.conf 
```

`$(prefix)` can be set by the installer, and so it can be an entirely different base directory, and then systemd will not find it. How is this more correct? Can you line one or two examples from other projects that use `$(prefix)/lib` instead of `$(libdir)`?

Reply:
The path selection is somewhat complicated. Ideally, we would use something like `$(shell pkg-config --variable=modules_load_dir systemd)` to get the path but there are considerations:

- Pros
  - The path will be correct.
  - Distros can override it using `PKG_CONFIG_SYSTEMD_MODULES_LOAD_DIR` environment variable.
- Cons
  - It will try to install to the system path even if user wants to install to a custom prefix for development.
  - It requires the `systemd.pc` file to be available.
    - It might potentially be part of a `devel` package not installed by default (not sure how common that is, Debian has it in `systemd`). 
    - When the project is built in chroot with explicit dependencies, `systemd` will need to be added just for finding the path.

One can usually rely on the pattern described in <https://www.bassi.io/articles/2018/03/15/pkg-config-and-paths/> to allow painless installation to development prefixes but that will not work here because [there are systems](https://github.com/systemd/systemd/blob/3abdd4bf2e7c6d7eef4e81b4a37fb5ebebc99ef7/meson_options.txt#L16-L17) where systemd’s `rootprefix` is different from `prefix`. So relying on `--define-variable=root_prefix=$(prefix)` would help for development but might break when installing to system.

> `$(prefix)` can be set by the installer, and so it can be an entirely different base directory, and then systemd will not find it. How is this more correct?

The same is true of `$(libdir)`. But using `$(prefix)/lib` is more correct because `systemd` [always uses `lib` subdirectory](https://github.com/systemd/systemd/blob/6b87254ad1ecb17e4ef88090c356705e55088c8a/src/core/systemd.pc.in#L77), even when `$(libdir)` is something else (e.g. `lib/x86_64-linux-gnu` on Debian).

Unfortunately, I do not think there is a nice procedure for finding the correct installation path. The closest would probably be trying to install to `$(shell pkg-config --variable=modules_load_dir systemd)` and if that fails on permission error, try `$(shell pkg-config --define-variable=root_prefix=$(prefix) --variable=modules_load_dir systemd)` but that might still cause a mess when one accidentally runs `sudo make install` instead of `make install`.

Reply:
Thanks for the detailed explanation, @jtojnar! This sounds quite complicated.

From what I can understand, using `$(libdir)` is at least more correct than hardcoding `/etc`. 

To get things even more correct, we need to use a conditional to get the most correct results. Would you be open to making a PR with what you think is better than using `$(libdir)`?